lib/s3middleware/lifecycleHelpers/LifecycleUtils.ts

```diff
@@ -98,6 +98,44 @@ export default class LifecycleUtils {
         });
     }
 
+    /**
+    * Find the most relevant trantition rule for the given transitions array
+    * and any previously stored transition from another rule.
+    * @param params - The function parameters
+    * @param params.transitions - Array of lifecycle rule transitions
```

Comment:
```suggestion
    * @param params.noncurrentTransitions - Array of lifecycle rule transitions
```
Did you mean this?
Also store and current date are not listed in the params.